static/js/components/PersonalTab.js

```diff
@@ -23,15 +39,74 @@ class PersonalTab extends React.Component {
     ui:             UIState,
     nextStep:       () => void,
     prevStep:       () => void,
+    dispatch:       Dispatch,
+    dashboard:      DashboardState,
+    enrollments: ProgramEnrollmentsState,
+    currentProgramEnrollment:    ProgramEnrollment,
   };
 
+  programListing = (programs: Program) => {
+    const makeMenuItems = R.map(program => (
+      <MenuItem value={program.id} key={program.id} primaryText={program.title} />
+    ));
+    const sortPrograms = R.sortBy(R.compose(R.toLower, R.prop('title')));
+    const menuItems = R.compose(makeMenuItems, sortPrograms);
+    return menuItems(programs || {}) ;
+  }
+
+  setProgramHelper = (event: Event, key: string, value: string) => {
+    const {
+      dispatch,
+      enrollments: { programEnrollments },
+    } = this.props;
+    let selected = programEnrollments.find(enrollment => enrollment.id === value);
+    dispatch(setCurrentProgramEnrollment(selected));
+  }
+
+  selectProgram = (setProgram: Function, program: Program) => {
+    const { dashboard: { programs } } = this.props;
+
+    return (
+      <SelectField
+        style={{width: "65%"}}
+        hintText={program ? program.title : "Select Program"}
+        onChange={setProgram}
+        value={program.id}
+        className="signup-dialog-select"
+      >
+        { this.programListing(programs) }
+      </SelectField>
+    );
+  }
+
   render() {
+    let currentId;
+    const {
+      currentProgramEnrollment,
+      enrollments: {programEnrollments},
+    } = this.props;
+
+    if (!_.isNil(currentProgramEnrollment)) {
+      currentId = currentProgramEnrollment.id;
+    }
+    let selectedProgram = programEnrollments.find(enrollment => enrollment.id === currentId);
+
     return (
       <div>
+        <Card shadow={1} className="program-select">
+          <div className="section-header">Which MicroMasters program are you signing up for?</div>
+          <br/>
+          { this.selectProgram(this.setProgramHelper, selectedProgram) }
+        </Card>
         <Card shadow={1} className="profile-form">
           <PersonalForm {...this.props} validator={personalValidation} />
           <ValidationAlert {...this.props} />
         </Card>
+        <div className="terms-of-service-container">
+          By clicking "Next" I certify that I agree with <a href="#">
+            MIT MicroMasters Terms of Service </a>. <br/> Read
+          our <a href="#">Nondiscrimination Policy</a>.
```

Comment:
This link also comes from `SignupDialog.js`: http://web.mit.edu/referencepubs/nondiscrimination/index.html

Please don't forget the `target="_blank"` here and above